### PR TITLE
It's our server skeleton. Please review code.

### PR DESCRIPTION
**Pull request by lyw0149**
@ethkim  @nlpenguin  @KaylaKwon @SarahhhC @djflexible 

below file is coverage test result.
[coverage.zip](https://github.com/sv-bootcamp/yoda/files/424339/coverage.zip)

### CODE REVIEW

File: src/server/controllers/users.controller.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export function getAll(req, res, next) {
+    User.find((err,doc) => {
+        if(err){
+            console.log(err)
+        }
+        res.send(doc)
+    })
+}
+
+export function index(req, res, next) {
```
Comment (ifndefdeadmau5) [inline src/server/controllers/users.controller.js]: Does this function shows user index page?
If it does, I think we should rename of the function name like 'getIndexPage' or something.
Because function name should be a verb as general programming style.

File: src/server/controllers/users.controller.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+'use struct'
+var User = require('mongoose').model('User')
```
Comment (ifndefdeadmau5) [inline src/server/controllers/users.controller.js]: I was told that variable name should be camel case and that upper case at first letter should use only when it is a class name.

File: .babelrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
+}
```
Comment (ethkim) [inline .babelrc]: I think this error refers to having an empty line at the end.

File: gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+const gulp			= require('gulp');
```
Comment (ethkim) [inline gulpfile.js]: Make indentation aligned with the rest of the lines.

File: src/server/config/db.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+'use strict'
+import mongoose from 'mongoose'
+
+var uri = 'mongodb://localhost/yoda'
+var options = {
+    server: {
+        poolSize: 20        // amount of connection in one instance
```
Comment (ethkim) [inline src/server/config/db.js]: To be absolutely anal, if you're putting comments inline, there should be precisely 2 spaces after the code and before the start of the comment. So it becomes:

       poolSize: 20  // amount of connection in one instance. 

That way the code looks cleaner overall. I'm not sure if this was mentioned in the styleguide. :-)

File: gulpfile.js (hunk 4 above)
Comment (ethkim) [inline gulpfile.js]: In fact, is it a js convention to add tabspaces before =? Otherwise we can just make it the usual var declarations:

const gulp = require('gulp');
const babel = require('gulp-babel');
const watch = require('gulp-nodemon');

..etc.

File: src/server/controllers/users.controller.js (hunk 2 above)
Comment (ethkim) [inline src/server/controllers/users.controller.js]: +1.

File: src/server/routes/users.route.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+'use strict'
+
+import express from 'express'
+import * as Users from '../controllers/users.controller'
+
+const router = express.Router()
+
+ router.get('/', Users.index)
+ //change to post
```
Comment (ethkim) [inline src/server/routes/users.route.js]: Not sure what this means. Is this referring to the next line?

File: src/test/node/test.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import test from 'tape';
+
+test('A passing test', (assert) => {
+
```
Comment (ethkim) [inline src/test/node/test.js]: No unnecessary whitespace line.

**Comment by ethkim** [comment]
Thanks Yangwoo! This looks good except a few minor nits. How do we ensure this doesn't break unit tests? I wonder if our unit tests are running in the background constantly. It'd be ideal if the "merge" operation is only possible when all unittests are passing. Otherwise we'll have to manually monitor that before each merge. 

Thanks!

**Comment by lyw0149** [comment]
First pull request is so dynamic. I think skeleton pull request could
except from unit test. Because it is just skeleton structure. I thought we
can start from here. Even unit-test. It`s my fault.
I will Re-pullrequest when unittest are pass.

2016. 8. 18. 오후 8:02에 "Ethan Kim" <notifications@github.com>님이 작성:

**Comment by ethkim** [comment]
No that's fine. I was just asking if there's a mechanism to ensure the
tests are passing at the time of merege.

This pull request looks good to me. :-) thanks!!

File: .babelrc (hunk 3 above)
Comment (nlpenguin) [inline .babelrc]: add an empty line

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#Useless Folders
+.c9/
+node_modules/
+dist-server/
+
+# DB Folder
+data/
+
+# scripts
+mongod
```
Comment (nlpenguin) [inline .gitignore]: add an empty line

File: gulpfile.js (hunk 4 above)
Comment (nlpenguin) [inline gulpfile.js]: I also hate tab-aligned `=`s.

File: gulpfile.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+const babel 		= require('gulp-babel');
+const watch 		= require('gulp-nodemon');
+const eslint		= require('gulp-eslint');
+const sourcemaps	= require('gulp-sourcemaps');
+const server		= require( 'gulp-develop-server');
```
Comment (nlpenguin) [inline gulpfile.js]: remove space after `(`

File: src/server/config/db.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    }
+}
+var db = mongoose.createConnection(uri, options)
+
+db.on('error', (err) => {                   // on: event listener
+	if(err){ console.error('db error', err) }
```
Comment (nlpenguin) [inline src/server/config/db.js]: indent with 2 spaces

File: src/server/config/db.js (hunk 5 above)
Comment (nlpenguin) [inline src/server/config/db.js]: also indents must be 2 spaces

File: src/server/config/express.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+import http             from 'http'
+import express          from 'express'
+import morgan           from 'morgan'
+import compress         from 'compression'
+import bodyParser       from 'body-parser'
+import methodOverride   from 'method-override'
+import users        from '../routes/users.route';
```
Comment (nlpenguin) [inline src/server/config/express.js]: do not tab align

File: src/server/config/express.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+export default function () {
+    var app = express();
```
Comment (nlpenguin) [inline src/server/config/express.js]: indent with 2 spaces

File: src/server/config/express.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	// Use the 'body-parser' and 'method-override' middleware functions
+	app.use(bodyParser.urlencoded({
+		extended: true
+	}));
+	app.use(bodyParser.json());
+	app.use(methodOverride());
+
+    app.use('/users',users)
```
Comment (nlpenguin) [inline src/server/config/express.js]: out of indentation.

File: src/server/models/users.model.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+    name: String,
+    age: Number,
+    field: String,
+    region: String,
+    skills: String,
+    regDate: { type: Date, default: Date.now() }
+})
+
+userSchema.set('toJSON', {
+	getters: true,      // bring all elements include virtuals
```
Comment (nlpenguin) [inline src/server/models/users.model.js]: tab with 2 spaces

**Comment by nlpenguin** [comment]
Overall, align everything with 2 spaces.

**Comment by nlpenguin** [comment]
I've added `LGTM` integration. You'd need someone else commenting `LGTM` before merging.

File: src/test/node/test.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+test('Assertions with tape.', (assert) => {
+  const expected = 'something to test';
+  const actual = 'something to test';
+
+  assert.equal(actual, expected,
+    'Given two mismatched values, .equal() should produce a nice bug report');
+
+  assert.end();
+});
```
Comment (nlpenguin) [inline src/test/node/test.js]: new line end

File: src/server/routes/users.route.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+ router.get('/insertMock', Users.saveTest)
+ router.get('/all', Users.getAll)
+
+export default router;
```
Comment (nlpenguin) [inline src/server/routes/users.route.js]: new line end

File: src/server/routes/users.route.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
+'use strict'
+
+import express from 'express'
+import * as Users from '../controllers/users.controller'
+
+const router = express.Router()
+
+ router.get('/', Users.index)
```
Comment (nlpenguin) [inline src/server/routes/users.route.js]: strange indent?

File: src/server/models/users.model.js [hunk 18]
```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+import mongoose from 'mongoose'
+const Schema = mongoose.Schema
+
+let userSchema = new Schema({
+    userId: String,
```
Comment (nlpenguin) [inline src/server/models/users.model.js]: 2 space indentation.

**Comment by nlpenguin** [comment]
If you're using intellij, you can auto-format all the files. What text editor are you using?

**Comment by nlpenguin** [comment]
@lyw0149  Use Sublime's reindent for all the files?:
screen shot 2016-08-18 at 7 09 34 pm

File: src/server/controllers/users.controller.js (hunk 2 above)
Comment (KaylaKwon) [inline src/server/controllers/users.controller.js]: @ifndefdeadmau5  even the collection name?

Comment (ifndefdeadmau5) [inline src/server/controllers/users.controller.js]: Hmm I'm not sure but it seems many people are using upper case at first letter for collection name.
